main.go

```diff
@@ -294,119 +248,41 @@ func main() {
 			setupLog.Error(err, "failed to start all-namespaces cache")
 			os.Exit(1)
 		}
 	}()
+
+	setupLog.Info("starting manager")
 	if err := mgr.Start(stopChannel); err != nil {
-		setupLog.Error(err, "Default manager exited non-zero")
+		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
-
-	setupLog.Info("Starting the Cmd.")
 }
 
 // newAllNamespacesClient creates a new client that watches (as opposed to the standard client) resources in all namespaces.
 // This client should be used only for resources and kinds that are retrieved from other namespaces than the watched one.
 // This will help keeping a reasonable memory usage for this operator since the cache won't store all other namespace scoped
 // resources (secrets, etc.).
-func newAllNamespacesClient(cfg *rest.Config) (client.Client, cache.Cache, error) {
-	// Create the mapper provider
-	mapper, err := apiutil.NewDynamicRESTMapper(cfg)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// Create the cache for the cached read client and registering informers
-	allNamespacesCache, err := cache.New(cfg, cache.Options{Scheme: scheme, Mapper: mapper, Namespace: ""})
+func newAllNamespacesClient(config *rest.Config) (client.Client, cache.Cache, error) {
+	clusterAllNamespaces, err := runtimecluster.New(config, func(clusterOptions *runtimecluster.Options) {
+		clusterOptions.Scheme = scheme
+	})
 	if err != nil {
 		return nil, nil, err
 	}
-	// Create the Client for Write operations.
-	c, err := client.New(cfg, client.Options{Scheme: scheme, Mapper: mapper})
-	if err != nil {
-		return nil, nil, err
-	}
-	// see https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.6/pkg/manager/manager.go#L374-L389
-	return &client.DelegatingClient{
-		Reader: &client.DelegatingReader{
-			CacheReader:  allNamespacesCache,
-			ClientReader: c,
-		},
-		Writer:       c,
-		StatusClient: c,
-	}, allNamespacesCache, nil
-
-}
-
-// addMetrics will create the Services and Service Monitors to allow the operator export the metrics by using
-// the Prometheus operator
-func addMetrics(ctx context.Context, cfg *rest.Config) {
-	// Get the namespace the operator is currently deployed in.
-	operatorNs, err := k8sutil.GetOperatorNamespace()
-	if err != nil {
-		if errors.Is(err, k8sutil.ErrRunLocal) {
-			setupLog.Info("Skipping CR metrics server creation; not running in a cluster.")
-			return
-		}
-	}
-
-	// Add to the below struct any other metrics ports you want to expose.
-	servicePorts := []v1.ServicePort{
-		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
-		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
-	}
-
-	// Create Service object to expose the metrics port(s).
-	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
```

Comment:
will the metric `service` and `servicemonitor` resources be created by the call to `ctrl.NewManager`?

Reply:
btw, you can check the resulting bundle manifests files in `bundle/` directory after running:
```
make bundle
```

Reply:
ah ok, cool. Thanks for the clarification!